api/src/main/java/net/kyori/adventure/util/Index.java

```diff
@@ -154,6 +157,37 @@ private Index(final Map<K, V> keyToValue, final Map<V, K> valueToKey) {
     return this.valueToKey.get(value);
   }
 
+  /**
+   * Gets the key for a value or throws an exception.
+   *
+   * @param value the value
+   * @return the key
+   * @throws NoSuchElementException if there is no key for the value
+   * @since 4.11.0
+   */
+  public @NotNull K keyOrThrow(final @NotNull V value) {
+    final K key = this.key(value);
+    if (key == null) {
+      throw new NoSuchElementException("There is no key for value " + value);
+    }
+    return key;
+  }
+
+  /**
+   * Gets a key by its value or returns
+   * a fallback key.
+   *
+   * @param value the value
+   * @param defaultKey the fallback key
+   * @return the key
+   * @since 4.11.0
+   */
+  @Contract("_, null -> null; _, !null -> !null")
+  public K keyOrDefault(final @NotNull V value, @Nullable K defaultKey) {
```

Comment:
```suggestion
  public K keyOr(final @NotNull V value, @Nullable K defaultKey) {
```

to match `ComponentSerializer.serializeOr`